### PR TITLE
Add thru-only de-embedding methods

### PR DESCRIPTION
**Pull request by keikawa**
I have added four thru-only de-embedding methods ('SplitPi', 'SplitTee', 'AdmittanceCancel', and 'ImpedanceCancel') to the de-embedding class. Test code and a tutorial are also included. Please check the code and give me some comments.

thanks.

### CODE REVIEW

File: doc/source/tutorials/Deembedding.ipynb [hunk 1]
```diff
@@ -150,33 +152,59 @@
...
+    "\n",
+    "![thru_dummy](figures/thru_dummy.svg)\n",
+    "\n",
+    "### Split methods\n",
+    "\n",
+    "These methods are based on lumped elements assumption as well as the open or short methods. The `SplitPi` and `SplitTee` methods assume that the embedding-network is Π-type and T-type as shown in the figure below. The values of the left and right parasitic components (i.e. Z and Y) are extracted from the thru dummy, and the DUT is de-embedded by multiplying the inverse matrix of the left and right from both sides of the test device. These methods can apply to only 2-port DUTs.\n",
+    "\n",
+    "![split_methods](figures/split_methods.svg)\n",
+    "\n",
+    "### Imittance cancel methods by swapping\n",
```
Comment (Ttl) [inline doc/source/tutorials/Deembedding.ipynb:180]: Immittance

File: skrf/calibration/deembedding.py [hunk 2]
```diff
@@ -510,3 +514,398 @@ def deembed(self, ntwk):
...
 
         return caled
+
+
+class SplitPi(Deembedding):
+    """
+    Remove shunt and series parasitics assumming pi-type embedding network.
+
+    A deembedding object is created with just one thru dummy measurement `dummy_thru`.
+    The thrh dummy is, for example, two test pads connected directly to each other.
```
Comment (Ttl) [inline skrf/calibration/deembedding.py:524]: thru

File: skrf/calibration/deembedding.py [hunk 3]
```diff
@@ -510,3 +514,398 @@ def deembed(self, ntwk):
...
+    References
+    ------------
+    ..  [4] L. Nan, K. Mouthaan, Y.-Z. Xiong, J. Shi, S. C. Rustagi, and B.-L. Ooi, 
+        “Experimental Characterization of the Effect of Metal Dummy Fills on Spiral Inductors,”
+        in 2007 IEEE Radio Frequency Integrated Circuits (RFIC) Symposium, Jun. 2007, pp. 307–310.
+
+    Example
+    --------
+    >>> import skrf as rf
+    >>> from skrf.calibration import AdmittanceCancel
```
Comment (Ttl) [inline skrf/calibration/deembedding.py:543]: Should import `SplitPi`

File: skrf/calibration/deembedding.py [hunk 4]
```diff
@@ -510,3 +514,398 @@ def deembed(self, ntwk):
...
+
+        return caled
+
+
+class SplitTee(Deembedding):
+    """
+    Remove series and shunt parasitics assumming tee-type embedding network.
+
+    A deembedding object is created with just one thru dummy measurement `dummy_thru`.
+    The thrh dummy is, for example, two test pads connected directly to each other.
```
Comment (Ttl) [inline skrf/calibration/deembedding.py:623]: thru

File: skrf/calibration/deembedding.py [hunk 5]
```diff
@@ -510,3 +514,398 @@ def deembed(self, ntwk):
...
+    References
+    ------------
+    ..  [5] M. J. Kobrinsky, S. Chakravarty, D. Jiao, M. C. Harmes, S. List, and M. Mazumder,
+        “Experimental validation of crosstalk simulations for on-chip interconnects using S-parameters,”
+        IEEE Transactions on Advanced Packaging, vol. 28, no. 1, pp. 57–62, Feb. 2005.
+
+    Example
+    --------
+    >>> import skrf as rf
+    >>> from skrf.calibration import AdmittanceCancel
```
Comment (Ttl) [inline skrf/calibration/deembedding.py:642]: `SplitTee`

File: skrf/calibration/deembedding.py [hunk 6]
```diff
@@ -510,3 +514,398 @@ def deembed(self, ntwk):
...
+        caled = left.inv ** ntwk ** right.inv
+
+        return caled
+
+
+class AdmittanceCancel(Deembedding):
+    """
+    Cancel shunt admittance by swapping (a.k.a Mangan's method).
+    A deembedding object is created with just one thru dummy measurement `dummy_thru`.
+    The thrh dummy is, for example, two test pads connected directly to each other.
```
Comment (Ttl) [inline skrf/calibration/deembedding.py:721]: thru

File: skrf/calibration/deembedding.py [hunk 7]
```diff
@@ -510,3 +514,398 @@ def deembed(self, ntwk):
...
+    ..  [3] S. Amakawa, K. Katayama, K. Takano, T. Yoshida, and M. Fujishima,
+        “Comparative analysis of on-chip transmission line de-embedding techniques,”
+        in 2015 IEEE International Symposium on Radio-Frequency Integration Technology,
+        Sendai, Japan, Aug. 2015, pp. 91–93.
+
+
+    Example
+    --------
+    >>> import skrf as rf
+    >>> from skrf.calibration import AdmittanceCancel
```
Comment (Ttl) [inline skrf/calibration/deembedding.py:840]: `ImpedanceCancel`

File: skrf/calibration/deembedding.py [hunk 8]
```diff
@@ -510,3 +514,398 @@ def deembed(self, ntwk):
...
+
+        return caled
+
+
+class ImpedanceCancel(Deembedding):
+    """
+    Cancel series impedance by swapping.
+
+    A deembedding object is created with just one thru dummy measurement `dummy_thru`.
+    The thrh dummy is, for example, two test pads connected directly to each other.
```
Comment (Ttl) [inline skrf/calibration/deembedding.py:819]: thru

File: skrf/calibration/tests/test_deembedding.py [hunk 9]
```diff
@@ -148,4 +237,40 @@ def test_shortopen(self):
         """
         dut = self.dm_so.deembed(self.raw2_1f)
         ind_calc = 1e9*np.imag(1/dut.y[0,0,0])/2/np.pi/dut.f
+        self.assertTrue(np.isclose(ind_calc, 1, rtol=self.rtol))
+
+    def test_spilitpi(self):
```
Comment (Ttl) [inline skrf/calibration/tests/test_deembedding.py:242]: `split`

File: skrf/calibration/tests/test_deembedding.py [hunk 10]
```diff
@@ -148,4 +237,40 @@ def test_shortopen(self):
...
+        """
+        After de-embedding, the network is a pure inductor of 1nH.
+        Test that this is true at a spot frequency.
+        """
+        dut = self.dm_pi.deembed(self.raw3_1f)
+        ind_calc = 1e9*np.imag(1/dut.y[0,0,0])/2/np.pi/dut.f
+        self.assertTrue(np.isclose(ind_calc, 1, rtol=self.rtol))
+
+    def test_spilittee(self):
```
Comment (Ttl) [inline skrf/calibration/tests/test_deembedding.py:251]: `split`

**Review by Ttl — COMMENTED** [review]
Some typoes I spotted.

**Comment by keikawa** [comment]
Thanks for your comment.
I've correctted typos.

**Comment by jhillairet** [comment]
Very nice addition and documentation, thank you! 

A few minor comments:
- in the section Through-only de-embedding, could you add the meaning of the big G and big S in your figure?
- could you make some references (like [1], [2] etc and a reference part at the end) in the documentation notebook to the reference you have used for `SplitPi`, `SplitTee`, `AdmittanceCancel`, and `ImpedanceCancel` methods (as you made nicely in the docstring)? This is useful for readers.
- if you think the image are a big too big, you can use HTML instead of Markdown to control the image size, like: ``. That's very up to you.

> Many other sophisticated methods of de-embedding have been reported in literature to extend the accuracy of DUT measurements up to high frequencies. While they can be done using the network operations of scikit-rf, their inclusion as a built-in de-embedding class within the package is welcomed as an open-source contribution to the project.

if you have some specific idea, please create an issue mentionning the methods (and references) that would be usefull to add in scikit-rf.


Otherwise I'm OK for merging

**Review by jhillairet — APPROVED** [review]

**Comment by keikawa** [comment]
Thank you for the review.
I have made the following changes for clarity.

- Changed the example of a thru dummy from an illustration to a micrograph.
- Revised the wording.
- Listed the references.
- Combined figure (c) with figure (a) and (b).

**Review by jhillairet — APPROVED** [review]

**Comment by jhillairet** [comment]
Thank you. Is it ready to be merged?

**Comment by keikawa** [comment]
Yes, please!

**Comment by jhillairet** [comment]
thank you!

**Pull request merged**